vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataType.java

```diff
@@ -46,85 +48,85 @@
-  CIRCLE_ARRAY(719, true, Circle[].class),
-  HSTORE(33670, true, Object.class),
-  OID(26, true, Object.class),
-  OID_ARRAY(1028, true, Object[].class),
-  VOID(2278, true, Object.class),
-  UNKNOWN(705, false, String.class),
-  TS_VECTOR(3614, false, String.class),
-  TS_VECTOR_ARRAY(3643, false, String[].class),
-  TS_QUERY(3615, false,  String.class),
-  TS_QUERY_ARRAY(3645, false,  String[].class);
+  BOOL(16, true, Boolean.class, JDBCType.BIT),
+  BOOL_ARRAY(1000, true, Boolean[].class, JDBCType.BIT),
+  INT2(21, true, Short.class, Number.class, JDBCType.SMALLINT),
+  INT2_ARRAY(1005, true, Short[].class, Number[].class, JDBCType.SMALLINT),
+  INT4(23, true, Integer.class, Number.class, JDBCType.INTEGER),
+  INT4_ARRAY(1007, true, Integer[].class, Number[].class, JDBCType.INTEGER),
+  INT8(20, true, Long.class, Number.class, JDBCType.BIGINT),
+  INT8_ARRAY(1016, true, Long[].class, Number[].class, JDBCType.BIGINT),
+  FLOAT4(700, true, Float.class, Number.class, JDBCType.REAL),
+  FLOAT4_ARRAY(1021, true, Float[].class, Number[].class, JDBCType.REAL),
+  FLOAT8(701, true, Double.class, Number.class, JDBCType.DOUBLE),
+  FLOAT8_ARRAY(1022, true, Double[].class, Number[].class, JDBCType.DOUBLE),
+  NUMERIC(1700, false, Numeric.class, Number.class, JDBCType.NUMERIC),
+  NUMERIC_ARRAY(1231, false, Numeric[].class, Number[].class, JDBCType.NUMERIC),
+  MONEY(790, true, Object.class, null),
+  MONEY_ARRAY(791, true, Object[].class, null),
+  BIT(1560, true, Object.class, JDBCType.OTHER),
+  BIT_ARRAY(1561, true, Object[].class, JDBCType.OTHER),
+  VARBIT(1562, true, Object.class, JDBCType.OTHER),
+  VARBIT_ARRAY(1563, true, Object[].class, JDBCType.OTHER),
+  CHAR(18, true, String.class, JDBCType.VARCHAR),
+  CHAR_ARRAY(1002, true, String[].class, JDBCType.VARCHAR),
+  VARCHAR(1043, true, String.class, JDBCType.VARCHAR),
+  VARCHAR_ARRAY(1015, true, String[].class, JDBCType.VARCHAR),
+  BPCHAR(1042, true, String.class, JDBCType.VARCHAR),
+  BPCHAR_ARRAY(1014, true, String[].class, JDBCType.VARCHAR),
+  TEXT(25, true, String.class, JDBCType.VARCHAR),
+  TEXT_ARRAY(1009, true, String[].class, JDBCType.VARCHAR),
+  NAME(19, true, String.class, JDBCType.VARCHAR),
+  NAME_ARRAY(1003, true, String[].class, JDBCType.VARCHAR),
+  DATE(1082, true, LocalDate.class, JDBCType.DATE),
+  DATE_ARRAY(1182, true, LocalDate[].class, JDBCType.DATE),
+  TIME(1083, true, LocalTime.class, JDBCType.DATE),
+  TIME_ARRAY(1183, true, LocalTime[].class, JDBCType.DATE),
+  TIMETZ(1266, true, OffsetTime.class, JDBCType.DATE),
+  TIMETZ_ARRAY(1270, true, OffsetTime[].class, JDBCType.DATE),
+  TIMESTAMP(1114, true, LocalDateTime.class, JDBCType.DATE),
+  TIMESTAMP_ARRAY(1115, true, LocalDateTime[].class, JDBCType.DATE),
+  TIMESTAMPTZ(1184, true, OffsetDateTime.class, JDBCType.DATE),
+  TIMESTAMPTZ_ARRAY(1185, true, OffsetDateTime[].class, JDBCType.DATE),
+  INTERVAL(1186, true, Interval.class, JDBCType.DATE),
+  INTERVAL_ARRAY(1187, true, Interval[].class, JDBCType.DATE),
```

Comment:
Some of the above mappings don't look right to me. `TIME` should be `TIME`, `TIMESTAMP`  should be `TIMESTAMP`, and `TIMESTAMPTZ` should probably be `TIMESTAMP_WITH_TIMEZONE` (but I'm not sure about that, we need to check).

And `INTERVAL` I'm not sure about but I'm pretty sure it's not a `DATE`.

Reply:
And `TIMETZ` is `TIME_WITH_TIMEZONE`.